diff --git a/lib/iconBase.d.ts b/lib/iconBase.d.ts
index 9891a022c2be4d36ec8f7f80a0f5380023959e5e..7e0440977dd52ae22320d5bb6a831f9d5ad7a205 100644
--- a/lib/iconBase.d.ts
+++ b/lib/iconBase.d.ts
@@ -12,8 +12,11 @@ export interface IconBaseProps extends React.SVGAttributes<SVGElement> {
     size?: string | number;
     color?: string;
     title?: string;
+    // FIXME: Added via pnpm patch to resolve missing className errors in React 19 / strict module resolution.
+    // Can be safely removed once react-icons explicitly adds className to IconBaseProps.
+    className?: string;
 }
 export type IconType = (props: IconBaseProps) => React.ReactNode;
 export declare function IconBase(props: IconBaseProps & {
     attr?: Record<string, string>;
-}): JSX.Element;
+}): React.JSX.Element;
